IMPLY-12348: Update description of UNION ALL in SQL syntax doc (#12710)
apache/druid: docs/querying/sql.md

## UNION ALL

> The UNION ALL operator fuses multiple queries together. Druid SQL supports the UNION ALL operator in two situations: top-level and table-level, as described below. Queries that use UNION ALL in any other way will not execute.

Reviewer: I wonder about the usage "fuses" here.

```suggestion
The UNION ALL operator fuses multiple queries together. Druid SQL supports the UNION ALL operator in two situations: top-level and table-level, as described below. Queries that use UNION ALL in any other way will fail.
```

### Top-level

> You can use UNION ALL at the very top outer layer of a SQL query - not in a subquery, and not in the FROM clause. The underlying queries run separately, back to back. Their results are concatenated together and appear one after the other.

Reviewer: ```suggestion
You can use UNION ALL at the very top outer layer of a SQL query - not in a subquery, and not in the FROM clause. The underlying queries run sequentially. Druid concatenates their results so that they appear one after the other.
```

SELECT COUNT(*) FROM tbl WHERE my_column = 'value2'
```

> > With top-level queries, you can't apply GROUP BY, ORDER BY, or any other operator to the results of a UNION ALL.

Reviewer: I wonder if the "no further processing" is is necessary. It does get lost in the revision.

Author: I think it's covered by "to the results".

Author: Updated doc with Ben's suggestions.
